packages/web-app-files/tests/unit/components/SideBar/SideBar.spec.js

```diff
@@ -0,0 +1,256 @@
+        const fileInfo = panel.find(selectors.fileInfo)
+        expect(fileInfo.exists()).toBeTruthy()
+
+        const body = panel.find(selectors.panelBody)
+        expect(body.exists()).toBeTruthy()
+
+        const navigation = panel.find(selectors.panelNavigation)
+        if (panels[expectedPanels[idx]].default) {
+          expect(body.text()).toContain(expectedPanels[idx] + ' content')
+
+          expect(navigation.exists()).toBeTruthy()
+
+          const navigationButtons = panel.findAll(selectors.panelNavigationButton)
+          expect(navigationButtons.length).toBe(2)
+          for (let i = 0; i < 2; i++) {
+            if (expectedPanels[i] === 'Details') {
+              continue
+            }
+            const buttonName = navigationButtons.at(i).text()
+            expect(expectedPanels).toContain(buttonName)
+          }
+        } else {
+          expect(body.text()).toBe('')
+          expect(navigation.exists()).toBeFalsy()
+        }
+      }
+    })
+
+    it('Opening a panel should work', async () => {
+      const wrapper = getWrapper(filesPersonalRoute, {
+        filename: 'testfile',
+        extension: 'jpg'
+      })
+
+      const buttons = {
+        Actions: selectors.actionButton,
+        People: selectors.peopleButton
+      }
+
+      const detailPanel = wrapper.find(panels.Details.selector)
+      expect(detailPanel.exists()).toBeTruthy()
+      expect(detailPanel.classes()).toContain('is-active')
+
+      for (const key in buttons) {
+        const button = wrapper.find(buttons[key])
+        expect(button.exists()).toBeTruthy()
+
+        let panel = wrapper.find(panels[key].selector)
+        expect(panel.exists()).toBeTruthy()
+
+        const body = panel.find(selectors.panelBody)
+        expect(body.text()).toBe('')
+
+        expect(panel.classes()).not.toContain('is-active')
+
+        await button.trigger('click')
+
+        panel = wrapper.find(panels[key].selector)
+        expect(panel.classes()).toContain('is-active')
+        expect(body.text()).toContain(key + ' content')
+      }
+    })
+  })
+})
+
+function getResource({ filename, extension, type }) {
+  // convert date to format 'Mon, 12 Jul 2021 11:04:33 GMT'
+  const d = new Date()
+  d.setDate(d.getDate() - 3)
+  const parts = d.toString().split(' ')
+  const mdate = `${parts[0]}, ${parts[2]} ${parts[1]} ${parts[3]} ${parts[4]} GMT` // ${parts[5]}`
+  return {
+    id: '4',
+    fileId: '4',
+    icon: type,
+    name: type === 'file' ? `${filename}.${extension}` : filename,
+    extension: extension,
+    path: type === 'file' ? `/${filename}.${extension}` : `/${filename}`,
+    type,
+    mdate,
+    size: '163',
+    indicators: [],
+    permissions: 'RDNVW',
+    starred: false,
+    etag: '"89128c0e8122002db57bd19c9ec33004"',
+    shareTypes: [],
+    ownerDisplayName: 'user1',
+    ownerId: 'user1',
+    canDownload: () => true,
+    isReceivedShare: () => true,
+    canBeDeleted: () => true,
+    canRename: () => true
```

Comment:
use only required fields. imo, every field need not be placed here which just increases the file length (makes test look more complex).